### PR TITLE
mrc-1191 translations

### PR DESCRIPTION
**Pull request by hillalex**
This creates a new component `Translated.vue`, and a dropdown in the header to toggle the language. It wraps 2 pieces of text in the new component as a demonstration.

It's quite a lot of lines changed (although 130 of them are just adding the translations) so let me know if you'd rather I broke it up more. But I thought would be good to see the component in action with the dropdown, so you can see the language being changed and the text being translated in real time.

Nb this branch is called `mrc-1191_header`

### CODE REVIEW

File: src/app/static/src/app/components/header/DropDown.vue [hunk 1]
```diff
@@ -0,0 +1,47 @@
+<template>
```
Comment (hillalex) [inline src/app/static/src/app/components/header/DropDown.vue]: I factored out this component to be re-used by the file menu and the language menu.

File: src/app/static/src/app/components/Translated.vue [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+<script lang="ts">
+    import Vue from "vue";
+    import i18next from "i18next";
+    import {mapState} from "vuex";
+    import {RootState} from "../root";
+
+    export default Vue.extend({
+        name: "Translated",
+        props: ["value"],
```
Comment (hillalex) [inline src/app/static/src/app/components/Translated.vue]: I don't love "value" as the prop name here - it's really the key of the translations dictionary, but "key" is a reserved word. Any ideas?

File: src/app/static/src/tests/components/baseline/baseline.test.ts [hunk 3]
```diff
@@ -11,7 +11,6 @@ import ErrorAlert from "../../../app/components/ErrorAlert.vue";
 import LoadingSpinner from "../../../app/components/LoadingSpinner.vue";
 
 const localVue = createLocalVue();
-Vue.use(Vuex);
```
Comment (hillalex) [inline src/app/static/src/tests/components/baseline/baseline.test.ts]: Have just moved this shared setup code into `setup.ts`

File: src/app/static/src/tests/testHelpers.ts [hunk 4]
```diff
@@ -35,3 +38,23 @@ export function testUploadErrorCommitted(url: string,
         });
     });
 }
+
+export function expectAllMutationsDefined(mutationDefinitions: any, mutationTree: MutationTree<any>) {
```
Comment (hillalex) [inline src/app/static/src/tests/testHelpers.ts]: Consolidated the test helper classes into one file, this used to live in it's own file.

File: src/app/static/src/app/components/Translated.vue (hunk 2 above)
Comment (EmmaLRussell) [reply src/app/static/src/app/components/Translated.vue]: Yeah, that's a bit confusing. I think some variant of 'key' would be better than 'value'. 'stringKey'?

**Comment by hillalex** [comment]
> Seems like a really nice approach, very clear.
> 
> Language menu is rendering weirdly for me - always seems to be open. And changing language doesn't have any effect on the strings which should be translated - though I can see the ChangeLanguage mutations coming through in the console. All the stuff on the right of the UserHeader seems to have migrated westwards.
> 
> Screenshot from 2019-12-10 11-33-00
> 
> Seem to have lost the fjelltop and avenir logos!

is this def on the mrc-1191_header branch? there's another old mrc-1191 branch which wasn't working

**Comment by EmmaLRussell** [comment]
> > Seems like a really nice approach, very clear.
> > Language menu is rendering weirdly for me - always seems to be open. And changing language doesn't have any effect on the strings which should be translated - though I can see the ChangeLanguage mutations coming through in the console. All the stuff on the right of the UserHeader seems to have migrated westwards.
> > Screenshot from 2019-12-10 11-33-00
> > Seem to have lost the fjelltop and avenir logos!
> 
> is this def on the mrc-1191_header branch? there's another old mrc-1191 branch which wasn't working

Ha, I've done it again :woman_facepalming:

**Comment by EmmaLRussell** [comment]
Yes, sorry that's all fine, I'm an idiot. Just the 'value' thing to change, if you want to.

File: src/app/static/src/app/components/Translated.vue (hunk 2 above)
Comment (hillalex) [reply src/app/static/src/app/components/Translated.vue]: Cool, I'll go with textKey

**Pull request merged**